homeassistant/components/mqtt/vacuum/schema_state.py

```diff
@@ -193,21 +215,23 @@ def _setup_from_config(self, config):
             )
         }
 
-    def _update_state_attributes(self, payload):
+    def _update_state_attributes(self, payload: dict[str, Any]) -> None:
         """Update the entity state attributes."""
         self._state_attrs.update(payload)
         self._attr_fan_speed = self._state_attrs.get(FAN_SPEED, 0)
         self._attr_battery_level = max(0, min(100, self._state_attrs.get(BATTERY, 0)))
 
-    def _prepare_subscribe_topics(self):
+    def _prepare_subscribe_topics(self) -> None:
         """(Re)Subscribe to topics."""
-        topics = {}
+        topics: dict[str, Any] = {}
 
         @callback
         @log_messages(self.hass, self.entity_id)
-        def state_message_received(msg):
+        def state_message_received(msg: ReceiveMessage) -> None:
             """Handle state MQTT message."""
-            payload = json_loads(msg.payload)
+            raw_payload: Any = json_loads(msg.payload)
+            assert isinstance(raw_payload, dict)
+            payload: dict[str, Any] = raw_payload
```

Comment:
I don't think the assert adds anything here. If you are sure it's a dict, then you can just use:
```suggestion
            payload: dict[str, Any] = json_loads(msg.payload)
```